File: Daily Resources/Day07/DSA - Sliding Window Solutions/freq_of_most_freq.cpp
class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        long long total = 0;
        int left = 0, max_freq = 0;
        
        for (int right = 0; right < nums.size(); ++right) {
            total += nums[right];
            while ((long long)(right - left + 1) * nums[right] - total > k) {
                total -= nums[left];
                left++;
            }
            max_freq = max(max_freq, right - left + 1);
        }
        
        return max_freq;
    }
};

File: Daily Resources/Day07/DSA - Sliding Window Solutions/freq_of_most_freq.py
class Solution:
    def maxFrequency(self, nums, k):
        nums.sort()
        left = 0
        total = 0
        max_freq = 0
        
        for right in range(len(nums)):
            total += nums[right]
            # Check if the total operations to make all elements in window equal to nums[right] exceeds k
            while (right - left + 1) * nums[right] - total > k:
                total -= nums[left]
                left += 1
            max_freq = max(max_freq, right - left + 1)
        
        return max_freq
